plugins/kafka-records-storage/src/main/java/com/github/bsideup/liiklus/kafka/KafkaRecordsStorage.java

```diff
@@ -183,4 +136,64 @@ public int getPartition() {
             });
         });
     }
+
+    private PartitionSource toPartitionSource(
+            DefaultKafkaReceiver<ByteBuffer, ByteBuffer> kafkaReceiver,
+            Flux<Record> recordFlux,
+            ConcurrentHashMap<Integer, Processor<TopicPartition, TopicPartition>> revocations,
+            TopicPartition topicPartition
+    ) {
+        val partitionList = Arrays.asList(topicPartition);
+        val partitionNum = topicPartition.partition();
+
+        val requests = new AtomicLong();
+
+        val revocation = ReplayProcessor.<TopicPartition>create(1);
+        revocations.put(partitionNum, revocation);
+
+        return new PartitionSource() {
+
+            @Override
+            public int getPartition() {
+                return partitionNum;
+            }
+
+            @Override
+            public Publisher<Record> getPublisher() {
+                return recordFlux
+                        .filter(it -> it.getPartition() == partitionNum)
+                        .delayUntil(record -> {
+                            if (requests.decrementAndGet() < 0) {
+                                return kafkaReceiver.doOnConsumer(consumer -> {
+                                    if (requests.get() < 0) {
+                                        consumer.pause(partitionList);
+                                    }
+                                    return true;
+                                });
+                            } else {
+                                return Mono.empty();
+                            }
+                        })
+                        .doOnRequest(requested -> {
+                            if (requests.addAndGet(requested) > 0) {
+                                DefaultKafkaReceiverAccessor.resume(kafkaReceiver, topicPartition);
+                            }
+                        })
+                        .takeUntilOther(revocation)
+                        .doFinally(__ -> DefaultKafkaReceiverAccessor.pause(kafkaReceiver, topicPartition));
+            }
+
+            @Override
+            public CompletionStage<Void> seekTo(long position) {
+                return kafkaReceiver
+                        .doOnConsumer(consumer -> {
+                            // + 1 because in Kafka "seek" moves the cursor to the desired next record
```

Comment:
Will it make sense to rename the method to 'seekAfter' instead?

Reply:
Well, the public API is "seekTo", just in Kafka they have this "+1" API. 
Also, the rest of Liiklus uses "latest acked position" notation instead of "the offset to start reading from"